### ISSUE TITLE
SCM - Add option to automatically select open file repository in Source Control Repositories view

### ISSUE THREAD

**Issue by juanvalino** [microsoft/vscode#152653]
When you have a workspace with many repositories would be very useful to have an option to automatically select in the Source Control Repositories view the repository of the current file open. As suggestion the option can be properties:

```
Name: SCM > Repositories: Auto Reveal
Descroption: Controls whether the Source Control Repositories view should automatically reveal and select the repository of the files when opening them.
Variable: scm.repositories.autoReveal
``` 

By this way you can focus you work very fast in the SCM of the file you are modifying, allowing a configuration option that simulates the behavior of the `Atom` editor and making easier the migration of the Atom users (as me) to vscode.

Currently, working with many repositories is a mess because the user interface doesn't focus the SCM stuff in the repository where is contained the file you are editing or viewing. 

I think not having this feature is an entry barrier for users coming from `Atom` editor.

**Comment by VSCodeTriageBot** [comment]
:slightly_smiling_face: This feature request received a sufficient number of community upvotes and we moved it to our backlog. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by lentzi90** [comment]
A screenshot to show the current mess when working with multiple repos:
![Screenshot from 2022-10-18 09-09-02](https://user-images.githubusercontent.com/9117693/196357396-8dfa6dbe-fff4-4fd1-9224-600b6138a40c.png)

My suggestion for how to do this would be to always just show one repository at a time. Instead of a tree/list with repositories, just use a drop down (that would automatically switch also when opening/focusing files in the editor). Having one action button and one message field for each repository in the view just seems strange and it takes up a lot of space.

Please take inspiration from the simplicity Atom provided on this:

![Screenshot from 2022-10-18 09-15-02](https://user-images.githubusercontent.com/9117693/196357287-3629fc80-3c5a-4269-ba6e-04b082e152d8.png)

**Comment by mvcoding** [comment]
Any news or update on this? I also switched from Atom to VS Code and really miss this feature here.

**Comment by vasil-sokolov** [comment]
Also trying to find a solution to this problem. The point is this always worked for me because I used the GitLense extension before. It implements such a behavior. But now I removed it and realized that this doesn't work out of the box in VS Code.
To be honest some repositories from my workspace are opened automatically. All of them are placed on level one. But others that are placed deeper VS Code doesn't reveal automatically.
